dhis-2/dhis-services/dhis-service-core/src/test/java/org/hisp/dhis/trackedentity/TrackedEntityInstanceStoreTest.java

```diff
@@ -223,19 +223,63 @@ void testQuery()
         TrackedEntityInstanceQueryParams params = new TrackedEntityInstanceQueryParams();
         List<TrackedEntityInstance> teis = teiStore.getTrackedEntityInstances( params );
         assertEquals( 6, teis.size() );
-        // Filter by attribute
+        // Filter by attribute with EQ
         params = new TrackedEntityInstanceQueryParams()
             .addFilter( new QueryItem( atA, QueryOperator.EQ, "Male", ValueType.TEXT, AggregationType.NONE, null ) );
         teis = teiStore.getTrackedEntityInstances( params );
         assertEquals( 2, teis.size() );
         assertTrue( teis.contains( teiD ) );
         assertTrue( teis.contains( teiE ) );
-        // Filter by attribute
+        // Filter by attribute with EQ
         params = new TrackedEntityInstanceQueryParams()
             .addFilter( new QueryItem( atA, QueryOperator.EQ, "Female", ValueType.TEXT, AggregationType.NONE, null ) );
         teis = teiStore.getTrackedEntityInstances( params );
         assertEquals( 1, teis.size() );
         assertTrue( teis.contains( teiF ) );
+
+        // Filter by attribute with STARTS
```

Comment:
I think these should each get their own test.

Currently, if `            .addFilter( new QueryItem( atA, QueryOperator.SW, "ma", ValueType.TEXT, AggregationType.NONE, null ) );` test case fails, the ones below that might work fine, will not run.

Reply:
I did think about it. But since there were many types of assertions already in the same test method, decided to add the new ones there. I will change it to its own test.

Reply:
are they now maybe duplicated 🤔 I see EQ, SW, EW in here and separate SW, EW tests below

Reply:
You can always adopt parametrized test approach for tests like these :) we now use junit5, so it's easy.